i18n-helpers/src/directives.rs

```diff
@@ -0,0 +1,130 @@
+use regex::{escape, Regex};
+
+#[derive(Debug, PartialEq)]
+pub enum Directive {
+    Skip,
+    TranslatorComment(String),
+}
+
+pub struct DirectiveFinder {
+    pattern_re: Regex,
+}
+
+impl Default for DirectiveFinder {
+    fn default() -> Self {
+        Self::new("i18")
+    }
+}
+
+impl DirectiveFinder {
+    pub fn new(prefix: &str) -> Self {
+        let pattern: String = r"<!-{2,}\s*".to_owned()
+            + &escape(prefix)
+            + &r"\s*[:-]"
+            + &r"(.*[^-])"  // the command part of the prefix, which must not end in a hyphen.
+            + &r"-{2,}>";
+        Self {
+            pattern_re: Regex::new(&pattern).expect("well-formed regex"),
+        }
+    }
+
+    pub fn find(&self, html: &str) -> Option<Directive> {
+        let captures = self.pattern_re.captures(html.trim())?;
+
+        let command = captures.get(1)?.as_str().trim();
+        match command.split(is_delimiter).next() {
```

Comment:
It feels like this could be handled by a (named) group in the regular expression?

Reply:
Sounds good: switching to a named group.